python/sglang/srt/model_executor/cuda_graph_runner.py

```diff
@@ -348,7 +348,13 @@ def capture_one_batch_size(self, bs: int, forward: Callable):
             spec_algorithm=self.model_runner.spec_algorithm,
             spec_info=spec_info,
             capture_hidden_mode=(
-                spec_info.capture_hidden_mode if spec_info else CaptureHiddenMode.NULL
+                CaptureHiddenMode.FULL
+                if self.model_runner.server_args.return_hidden_states
+                else (
+                    spec_info.capture_hidden_mode
+                    if spec_info
+                    else CaptureHiddenMode.NULL
+                )
```

Comment:
@zhaochenyang20 What I meant is here. It seems like it is necessary for the `capture_hidden_mode` to be known at engine init time. Otherwise, the decode cuda graph will not contain the return hidden state logic and this cant be changed by sampling args.